src/components/ui/Modal.tsx

```diff
@@ -0,0 +1,85 @@
+import classNames from "classnames";
+import React, { Fragment } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+
+interface ModalProps {
+  size?: "lg" | "sm";
+  dialogClassName?: string;
+
+  show?: boolean;
+  fullscreen?: boolean;
+  backdrop?: boolean | "static";
+
+  onHide?: () => void;
+  onExited?: () => void;
+  onShow?: () => void;
+}
+
+export const Modal: React.FC<ModalProps> = ({
+  children,
+  show,
+  onHide,
+  onExited,
+  onShow,
+  size,
+  fullscreen,
+  dialogClassName,
+  backdrop = true,
+}) => {
+  return (
+    <Transition appear show={show} as={Fragment}>
+      <Dialog
+        as="div"
+        className="fixed inset-0 flex min-h-full items-center justify-center z-50"
```

Comment:
Bootstrap was sitting at 1050, now Modal is sitting at `z-50` just above the Hud at `z-40`. Anima/Synaps sits at 1000.